common/src/main/java/io/pravega/common/hash/HashHelper.java

```diff
@@ -160,4 +166,77 @@ public int getAsInt() {
             return getHashCode().asInt();
         }
     }
+    
+    /**
+     * Hashes a bufferview in-place. (in contrast to murmur above which needs to copy data into a single array)
+     * (This passes the smhasher quality test suite)
+     *
+     * @param bufferView The input.
+     */
+    public static final long hashBufferView(BufferView bufferView) {
+        long multiple = 6364136223846793005L; // From knuth
+        long inc = 1442695040888963407L; // From knuth
+        long state = 0xc3a5c85c97cb3127L; // (arbitrary, from farmhash) 
+        long wheil = 0x9ae16a3b2f90404fL; // (arbitrary, from farmhash) 
```

Comment:
It's not a typo. "Wheil" is a person's last name. A "Wheil counter" is a counter that instead of being incremented by one is incremented by some other odd number. It will still take 2^64 increments to return to the original value but gets there in a more complicated way.

Reply:
is there any reference to Wheil Counter?(a quick search on the term didnt show up any results)

Reply:
@abhinb Ah, yes. That's because it did misspell it ;-) 
Here you go: https://en.wikipedia.org/wiki/Weyl_sequence